### ISSUE TITLE
[pickers] Weekdays start from Monday with `en-US` with `AdapterLuxon`

### ISSUE THREAD

**Issue by matan-atiya** [mui/mui-x#7670]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Steps to reproduce 🕹

Link to live example:

Steps:
1. install the latest version of mui
2. use one of the date pickers 
3. Date picker start from Monday

### Current behavior 😯

I installed the latest version and used a couple of date pickers - all the date pickers start from Monday instead of Sunday.

LocalizationProvider dateAdapter is config to AdapterLuxon.

![Screenshot 2023-01-23 at 12 23 56](https://user-images.githubusercontent.com/92845991/214016737-d9fbe6a0-86f8-4351-a1af-cb0afea11561.png)

<img width="303" alt="Screenshot 2023-01-23 at 12 24 58" src="https://user-images.githubusercontent.com/92845991/214016968-57d980b5-badf-4031-a446-45dcc73ce257.png">

### Expected behavior 🤔

Date picker suppose to start from Sunday

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by LukasTy** [comment]
Thank you for opening this issue, but I'm afraid that we can't do anything on this regard except suggest changing the date management library if this behavior is strictly necessary or consider waiting until the [localized weekdays are supported by Luxon](https://github.com/mui/mui-x/pull/7646/files#diff-1a52cf4cdf6576b7d5a5e3ba04d1135846dab1b00bb57835ae36a7af2906074aR183) ([Intl](https://github.com/moment/luxon/issues/373#issuecomment-885032455)).

**Comment by LukasTy** [comment]
> Thanks for the answer, what date management library is most compatible to work with MUI?

[On v6 (next) documentation](https://github.com/mui/mui-x/issues/7670#issuecomment-1400522693) we endorse using `dayjs` and are using it's adapter on demos. 😉

**Comment by matan-atiya** [comment]
When I try to use AdapterDayjs is getting the next errors - although I have installed @date-io/dayjs

**Comment by matan-atiya** [comment]
Thanks will migrate my code to that library

**Comment by matan-atiya** [comment]
After trying to migrate I saw I have to change a lot of code in my app so in the end, I manage to create a custom Luxon adapter that is working great - for languages that start from Sunday it will change the week order for all the others he continues with the normal behavior of Luxon (start from Monday).

Used both threads to create the adapter - 

https://stackoverflow.com/questions/53382465/how-can-i-determine-if-week-starts-on-monday-or-sunday-based-on-locale-in-pure-j

https://github.com/mui/material-ui-pickers/issues/1270#issuecomment-567075962

Hope that will help others that encounter the same problem - 


```
import { AdapterLuxon } from "@mui/x-date-pickers/AdapterLuxon";
import { DateTime, Info } from "luxon";

const weekStart = (region: string, language: string) => {
  const regionSat = "AEAFBHDJDZEGIQIRJOKWLYOMQASDSY".match(/../g);
  const regionSun =
    "AGARASAUBDBRBSBTBWBZCACNCODMDOETGTGUHKHNIDILINJMJPKEKHKRLAMHMMMOMTMXMZNINPPAPEPHPKPRPTPYSASGSVTHTTTWUMUSVEVIWSYEZAZW".match(
      /../g
    );
  const languageSat = ["ar", "arq", "arz", "fa"];
  const languageSun =
    "amasbndzengnguhehiidjajvkmknkolomhmlmrmtmyneomorpapssdsmsnsutatethtnurzhzu".match(
      /../g
    );

  return region
    ? regionSun && regionSun.includes(region)
      ? "sun"
      : regionSat && regionSat.includes(region)
      ? "sat"
      : "mon"
    : languageSun && languageSun.includes(language)
    ? "sun"
    : languageSat.includes(language)
    ? "sat"
    : "mon";
};

const weekStartLocale = (locale: string) => {
  const parts = locale.match(
    /^([a-z]{2,3})(?:-([a-z]{3})(?=$|-))?(?:-([a-z]{4})(?=$|-))?(?:-([a-z]{2}|\d{3})(?=$|-))?/i
  );
  return weekStart((parts && parts[4]) || "", (parts && parts[1]) || "");
};

export class CustomLuxonAdapter extends AdapterLuxon {
  getWeekdays = () => {
    // need to copy the existing, and use Info to preserve localization
    const days = [...Info.weekdaysFormat("narrow", { locale: this.locale })];
    const sundayStart = weekStartLocale(this.locale) === "sun";
    if (sundayStart) {
      // remove Sun from end of list and move to start of list
      days.unshift(days.pop() || "");
    }
    return days;
  };

  getWeekArray = (date: DateTime) => {
    const sundayStart = weekStartLocale(this.locale) === "sun";
    const endDate = date
      .endOf("month")
      // if a month ends on sunday, luxon will consider it already the end of the week
      // but we need to get the _entire_ next week to properly lay that out
      // so we add one more day to cover that before getting the end of the week
      .plus({ days: sundayStart ? 1 : 0 })
      .endOf("week");
    const startDate = date
      .startOf("month")
      .startOf("week")
      // must subtract 1, because startOf('week') will be Mon, but we want weeks to start on Sun
      // this is the basis for every day in a our calendar
      .minus({ days: sundayStart ? 1 : 0 });

    const { days } = endDate.diff(startDate, "days").toObject();

    const weeks: DateTime[][] = [];
    new Array(Math.round(days || 0))
      .fill(0)
      .map((_, i) => i)
      .map((day) => startDate.plus({ days: day }))
      .forEach((v, i) => {
        if (i === 0 || (i % 7 === 0 && i > 6)) {
          weeks.push([v]);
          return;
        }

        weeks[weeks.length - 1].push(v);
      });

    // a consequence of all this shifting back/forth 1 day is that you might end up with a week
    // where all the days are actually in the previous or next month.
    // this happens when the first day of the month is Sunday (Dec 2019 or Mar 2020 are examples)
    // or the last day of the month is Sunday (May 2020 or Jan 2021 is one example)
    // so we're only including weeks where ANY day is in the correct month to handle that
    return weeks.filter((w) => w.some((d) => d.month === date.month));
  };
}


```

**Comment by dmarcucci** [comment]
@matan-atiya Thanks for this!

**Comment by flaviendelangle** [comment]
Luxon now correctly support localized week.

The clean implementation will be merged in the v7 alpha branch (#10964) and I don't know yet if we will cherry-pick it on v6.

You can apply the exact same behavior [like on this codesandbox](https://codesandbox.io/s/date-and-time-pickers-forked-ffkc87?file=/src/App.tsx).
You just need Luxon 3.4.4 or above :+1: 
This should be more stable than the custom implementation proposed above :+1:
